examples/meta/src/features/string_char.sg

```diff
@@ -0,0 +1,10 @@
+File words = csv_file("../../data/words.dat")
+
+#![create_features]
+Features f = string_features(words, enum EAlphabet.RAWBYTE)
+#![create_features]
+
```

Comment:
I am stuck. I am trying to make get_features work. However, it is overloaded and it is used in many undocumented python examples. 

https://github.com/shogun-toolbox/shogun/blob/1c9ac30237f8b9ba5b2f9fe619fa9fabd7ee361f/src/shogun/features/StringFeatures.cpp#L1022

https://github.com/shogun-toolbox/shogun/blob/1c9ac30237f8b9ba5b2f9fe619fa9fabd7ee361f/src/shogun/features/StringFeatures.cpp#L1030

https://github.com/shogun-toolbox/shogun/blob/1c9ac30237f8b9ba5b2f9fe619fa9fabd7ee361f/src/shogun/features/StringFeatures.cpp#L1062

I don't want to put them in the base interface because they are specific to string features.

You said earlier  
> The way around this is to communicate with the classes via put get (to set parameters) and then call helper methods.

put and get for watched parameters is fine but what helper methods you mean?

Reply:
I just spend quite some time trying to register `get_features` with `watch_param`. Tricky one!

You will first need to make the method `const` (both of them as one calls the other). The compiler error you see when trying to register it comes from const, not from overloading (although it looks like that).

Then you will hit another compiler error, which complains about `bool shogun::any_detail::compare` not matching with the call (among others). This comes from the fact that anything that is ever any'ed in Shogun needs to be comparable and cloneable for `SGObject::clone/equals`. This hasn't yet been done for `SGStringList`. You will have to add `equals` and `clone` to `SGStringList` (see SGMatrix/SGVector/ for inspiration). Then those compiler errors should vanish. I don't know what happens then, but it will bring you one step closer to making the `get` work for the list of string features.